### DISCUSSION TITLE
Idea: Virtual scrolling on remote data source

### DISCUSSION THREAD

**Discussion by acetinick** [orchidjs/tom-select#54] [Ideas]
Would be really cool to be able to have feature to have virtual scrolling when select is open to enable pagination via virtual scrolling.

An example from Syncfusion library.
https://ej2.syncfusion.com/documentation/drop-down-list/how-to/achieve-virtual-scrolling/

**Comment by oyejorge** [comment]
I thought this could be done without modifying the core but I might be wrong...

**Comment by oyejorge** [comment]
Ok, making progress: https://github.com/orchidjs/tom-select/blob/master/test/html/virtual-scroll.html

Note, this will require version 1.5 which will be released soon

**Reply by acetinick** [reply]
Wow very cool) keen to play with once its ready!

**Comment by oyejorge** [comment]
Reworked that first example a bit and just released 1.7.0 with the [virtual_scroll plugin](https://tom-select.js.org/plugins/virtual_scroll/)

**Reply by yanivtoledano** [reply]
Thanks for this!!

Question: everytime it triggers a fetch of the next url scroll gets reset to the start of the list. Any way to disable this such that scroll position is unchanged between fetches of urls?

**Comment by raysuelzer** [comment]
Is there a way to do this with local data like select2 and selectize? I.E. The expense is rarely in fetching select options from the server, i can request 100k items in 100ms but rendering is the actual issue? I don't see an option on the virtual_scroll plugin to handle the case where i have the data locally, but rendering it all at once is not feasible?
